### ISSUE TITLE
Documentation and code are prone to desynchronization

### ISSUE THREAD

**Issue by acinonyx** [satnogs/satnogs-docs#19]
While a central repository for all documentation (like this one) seems initially as a good idea, there are many disadvantages when it documents code from multiple repositories. A compatibility matrix is then required that will match the actual source code version with the documentation version. Normally, only a matrix of components that assemble a functional whole system (integration) should be required with some generic/common documentation.

Keeping component specific documentation on the same repository as the source code, eliminates this need and keeps it always in-sync since software and documentation can be updated into a single pull request.

'satnogs-docs' still has a purpose; to keep generic and common documentation that is system-design specific. Taking into account that RTD is used for rendering documentation, this can only be achieved by using submodules. These submodules will pull specific releases or branches of all components and create the whole documentation. Additionally, two branches can be maintained; one for what we consider stable releases (stable) and one for work in progress (master).

**Comment by comzeradd** [comment]
Starting from the last point, afaik RTD can handle multiple branches so that should be fine.

The idea behind this repo was/is that RTD renders the documentation in a proper and nice looking manner for potential users and developers to read. Reading documentation through a code repository is not that helpful. If you think that using submodules can combine best elements of the two approaches, then let's do this.

One other reason we picked RTD is that it's sphinx based, so it's easy to programmatically generate documentation. I'm saying this because one option would be to have the documentation on our wiki, but this will make the desynchronization gap even bigger.

**Comment by acinonyx** [comment]
I agree, RTD is the way to go. 'satnogs-docs' will pull docs from the client and generate the pages which will be available at http://docs.satnogs.org/ . Nothing shall change from readers point of view.
